### PR TITLE
Add custom color picker

### PR DESCRIPTION
**Pull request by jehna**
Demo: 

https://github.com/raimohanska/ourboard/assets/2676795/8b3a91d9-62b6-4d9a-9e6e-68a1474b08f7

### CODE REVIEW

File: frontend/src/board/contextmenu/colors.tsx [hunk 1]
```diff
@@ -32,3 +45,10 @@ export function colorsSubMenu({ board, focusedItems, dispatch }: SubmenuProps) {
         dispatch({ action: "item.update", boardId: b.id, items: updated })
     }
 }
+
+function itemColorOrDefault(items: Item[]) {
+    if (items.length === 0) return YELLOW
+    const firstNote = items.find((i): i is Note => i.type === "note")
```
Comment (raimohanska) [inline frontend/src/board/contextmenu/colors.tsx:51]: Use isColoredItem instead, to support all colorable items

File: frontend/src/board/contextmenu/colors.tsx [hunk 2]
```diff
@@ -32,3 +45,10 @@ export function colorsSubMenu({ board, focusedItems, dispatch }: SubmenuProps) {
         dispatch({ action: "item.update", boardId: b.id, items: updated })
     }
 }
+
+function itemColorOrDefault(items: Item[]) {
+    if (items.length === 0) return YELLOW
```
Comment (raimohanska) [inline frontend/src/board/contextmenu/colors.tsx:50]: May I suggest introducing a DEFAULT_NOTE_COLOR constant to be used here as well as in `newNote` and `cutCopyPasteHandler` at least?

Comment (jehna) [reply frontend/src/board/contextmenu/colors.tsx]: Added!

File: frontend/src/board/contextmenu/colors.tsx (hunk 1 above)
Comment (jehna) [reply frontend/src/board/contextmenu/colors.tsx]: Changed!

**Comment by raimohanska** [comment]
Thanks! This is a great feature!

**Pull request merged**